monai/apps/detection/transforms/array.py

```diff
@@ -296,3 +305,61 @@ def __call__(  # type: ignore
         """
 
         return flip_boxes(boxes, spatial_size=spatial_size, flip_axes=self.spatial_axis)
+
+
+class ClipBoxToImage(Transform):
+    """
+    Clip the bounding boxes and the associated labels/scores to makes sure they are within the image.
+    There might be multiple arryas of labels/scores associated with one array of boxes.
+
+    Args:
+        remove_empty: whether to remove the boxes and corresponding labels that are actually empty
+    """
+
+    backend = [TransformBackends.TORCH, TransformBackends.NUMPY]
+
+    def __init__(self, remove_empty: bool = False) -> None:
+        self.remove_empty = remove_empty
+
+    def __call__(  # type: ignore
+        self,
+        boxes: NdarrayOrTensor,
+        labels: Union[Sequence[NdarrayOrTensor], NdarrayOrTensor],
+        spatial_size: Union[Sequence[int], int],
+    ) -> Tuple[NdarrayOrTensor, Tuple]:
+        """
+        Args:
+            boxes: bounding boxes, Nx4 or Nx6 torch tensor or ndarray. The box mode is assumed to be ``StandardMode``
+            labels: Sequence of int array. Each element represents classification labels or scores
+                corresponding to ``boxes``, sized (N,).
+            spatial_size: The spatial size of the image where the boxes are attached. len(spatial_size) should be in [2, 3].
+
+        Returns:
+            - clipped boxes, does not share memory with original boxes
+            - clipped labels, does not share memory with original labels
+
+        Example:
+            .. code-block:: python
+
+                boxes = torch.ones(2, 6)
+                class_labels = torch.Tensor([0, 1])
+                pred_scores = torch.Tensor([0.4, 0.5])
+                labels = (class_labels, pred_scores)
+                spatial_size = [32, 32, 32]
+        """
+        spatial_dims: int = get_spatial_dims(boxes=boxes)
+        spatial_size = ensure_tuple_rep(spatial_size, spatial_dims)  # match the spatial image dim
+
+        boxes_clip, keep = clip_boxes_to_image(boxes, spatial_size, self.remove_empty)
+
+        labels_tuple = ensure_tuple(labels)
+        labels_clip_list = []
+
+        keep_t: torch.Tensor = convert_data_type(keep, torch.Tensor)[0]
+        for i in range(len(labels_tuple)):
+            labels_t: torch.Tensor = convert_data_type(labels_tuple[i], torch.Tensor)[0]
+            if boxes.shape[0] != labels_t.shape[0]:
+                raise ValueError("boxes.shape[0] should be equal to the number of labels or scores.")
+            labels_t = deepcopy(labels_t)[keep_t, ...]
```

Comment:
@wyli I am not very sure whether `labels_t = deepcopy(labels_t[keep_t, ...])` would be better?

Thanks.